cosmic_ray/util.py

```diff
@@ -64,5 +64,7 @@ def build_mutations(ops, to_ops):
     return [
         (idx, to_op)
         for idx, from_op in enumerate(ops)
-        for to_op in to_ops(from_op)
+        # note: mutations to self are excluded.
+        # None is a special mutation meaning to delete the operator!
+        for to_op in to_ops(from_op) if to_op is None or not isinstance(from_op, to_op)
```

Comment:
There are two cases here:

1) to_op is not None so we do the isinstance() check to figure out whether or not to include the operator in the list of possible mutations
2) to_op is None. In this case isinstance(from_op, None) will blow up because the second parameter needs to be a class.

The `if to_op is None or isinstance(from_op, to_op)` expression handles both scenarios very elegantly. If 2) then the rest of the expression is not evaluated and None is returned. If 1) the left part of the expression is False so the next right part is evaluated. Since the left part of the expression has confirmed that to_op is not None then we're confident that the isinstance() method will always work.

Should I add the above explanation as a comment in the code? I though the 2 lines were sufficient but now I realize I could be more verbose.